internal/common/autogen/customtypes/object.go

```diff
@@ -129,6 +129,14 @@ type ObjectValueInterface interface {
 	ValuePtrAsAny(ctx context.Context) (any, diag.Diagnostics)
 }
 
+func (v ObjectValue[T]) ToTerraformValue(ctx context.Context) (tftypes.Value, error) {
+	if v.ObjectValue.Equal(basetypes.ObjectValue{}) {
```

Comment:
q: Is there a reason why this implementation deviates from other implementations which check for null? 

> if v.ElementType(ctx) == nil {